src/proto/streams/recv.rs

```diff
@@ -246,6 +250,19 @@ where
         Ok(())
     }
 
+    pub fn set_target_connection_window(&mut self, target: WindowSize, task: &mut Option<Task>) {
+        let current = self.flow.available() + self.in_flight_data;
+        if target > current {
```

Comment:
Now that I file this, it occurred to me that this won't really allow turning the target window down, only up. I'll go fix that.